internal/cnpi/plugin/operatorclient/client.go

```diff
@@ -97,10 +99,13 @@ func (e *extendedClient) Delete(
 	if err != nil {
 		return err
 	}
-	if err := e.Client.Patch(ctx, obj, client.MergeFrom(origObj)); err != nil {
-		contextLogger.Error(err, "while patching before delete")
-		return err
+	if !reflect.DeepEqual(origObj, obj) {
+		if err := e.Client.Patch(ctx, obj, client.MergeFrom(origObj)); err != nil && !apierrors.IsNotFound(err) {
+			contextLogger.Error(err, "while patching before delete")
+			return err
+		}
 	}
+
 	return e.Client.Delete(ctx, obj, opts...)
```

Comment:
Currently, it could happen that we execute a delete that we know doesn't exist anymore. IMO, this is fine, given that we preserve the original behavior of invoking a Delete with the runtime client.